client/src/main/java/io/pravega/client/stream/impl/ClientFactoryImpl.java

```diff
@@ -244,8 +244,10 @@ private Segment getSegmentForRevisionedClient(String scope, String streamName) {
 
     @Override
     public void close() {
-        controller.close();
+        // wait for 5 seconds before forcibly terminating the watermarkReader threads.
+        ExecutorServiceHelpers.shutdown(watermarkReaderThreads);
```

Comment:
nit: There is no timeout duration specified, unlike what the comment suggests. This comment might be referring on the default value used in the `shutdown` method. If someone changes that value, this comment would go stale. It might be useful rather to say something like `Wait for default timeout duration before...`

Reply:
updated it.